### PR TITLE
Feat/layergroup

### PR DESCRIPTION
**Pull request by amcgee**
This is a companion and pre-requisite PR to https://github.com/dhis2/maps-app/pull/117.

This change adds support for a generic Layer Group which can contain any other layer types.  It also fixes a few small issues found along the way.

GitHub is having trouble with the diffs, I'll try reverting stylistic changes

### CODE REVIEW

**Comment by amcgee** [comment]
The diffs are nicer now, just ignore the removed semicolons

File: src/layers/GeoJson.js [hunk 1]
```diff
@@ -23,44 +23,44 @@ export const GeoJson = L.GeoJSON.extend({
...
             pointToLayer: this.pointToLayer.bind(this),
             ...options,
-        });
+        })
     },
 
     addLayer(layer) {
-        const { label, hoverLabel } = this.options;
-        const { properties } = layer.feature;
+        const { label, hoverLabel, popup } = this.options
```
Comment (turban) [inline src/layers/GeoJson.js]: popup is not used

File: src/Map.js [hunk 2]
```diff
@@ -1,20 +1,15 @@
...
-import clientCluster from './layers/ClientCluster';
-import serverCluster from './layers/ServerCluster';
-import earthEngine from './layers/EarthEngine';
-import legend from './controls/Legend';
-import fitBounds from './controls/FitBounds';
-import search from './controls/Search';
-import measure from './controls/Measure';
-import { toLatLng, toLatLngBounds, toLngLatBounds } from './utils/geometry';
+import L from 'leaflet'
+import layerTypes from './layerTypes'
```
Comment (turban) [inline src/Map.js]: Nice!

**Review by turban — APPROVED** [review]
LGTM. Just remove the popup options :-)

**Pull request merged**